watch/watch.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package watch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage"
+)
+
+// Watcher allows for watches to be registed on queries. When a dependency of a watched
+// query changes, a notification will be send back to the user.
+type Watcher struct {
```

Comment:
Now that the storage layer doesn't require IDs in order to register triggers, the watch package has been refactored to separate the `watch.Watcher` (manages watches) from the `watch.Handle` (the actual watches).